build/mac/BUILD.gn

```diff
@@ -393,250 +255,191 @@ if (install_omaha4_with_pkg) {
+  outputs = [ "$packaging_dir/keystone_install.sh" ]
+}
 
-      if (sparkle_eddsa_private_key != "") {
-        deps += [ ":generate_eddsa_sig_for_dmg" ]
-      }
+copy("dmg_background_image") {
+  sources = [ "dmg-background.png" ]
+  outputs = [ "$packaging_dir/chrome_dmg_background.png" ]
+}
 
-      if (build_delta_installer && last_chrome_installer != "") {
-        deps += [ ":build_delta_installer" ]
-        if (sparkle_eddsa_private_key != "") {
-          deps += [ ":generate_eddsa_sig_for_delta" ]
-        }
-      }
-    }
+copy("dmg_icon") {
+  dmg_icon_file_name = "dmg"
+  if (!is_release_channel) {
+    dmg_icon_file_name = "dmg-$brave_channel"
   }
+  sources = [ "$dmg_icon_file_name.icns" ]
+  outputs = [ "$packaging_dir/chrome_dmg_icon.icns" ]
 }
 
-copy("copy_notarize_script") {
-  sources = [ "notarize_dmg_pkg.py" ]
+copy("ds_store") {
+  ds_store_file_name = "DS_Store"
+  if (!is_release_channel) {
+    ds_store_file_name = "DS_Store.$brave_channel"
+  }
+  sources = [ "DS_Store-files/$ds_store_file_name" ]
+  outputs = [ "$packaging_dir/chrome_dmg_dsstore" ]
+}
 
+copy("helper_entitlements") {
+  visibility = [ ":package" ]
+  sources = [ "//brave/app/helper-entitlements.plist" ]
+  public_deps = [ "//chrome/installer/mac" ]
   outputs = [ "$packaging_dir/{{source_file_part}}" ]
 }
 
-action("notarize_dmg_pkg") {
-  script = "$packaging_dir/notarize_dmg_pkg.py"
-
-  if (is_official_build) {
-    development_arg = ""
+if (is_official_build) {
+  copy("provisionprofile") {
     if (is_release_channel) {
-      provisioning_profile = "//brave/build/mac/release.provisionprofile"
+      channel_nonempty = "release"
     } else {
-      provisioning_profile =
-          "//brave/build/mac/${brave_channel}.provisionprofile"
+      channel_nonempty = brave_channel
     }
-  } else {
-    development_arg = "--development"
-    provisioning_profile = "//brave/build/mac/dummy.provisionprofile"
+    sources = [ "${channel_nonempty}.provisionprofile" ]
+    outputs = [ "${packaging_dir}/${channel_nonempty}.9c57a508dc89402964eb013d30ef6df4da849aa9.provisionprofile" ]
   }
+}
 
-  args = [
-    "--identity",
-    mac_signing_identifier,
-    string_join("=",
-                [
-                  "--notary-arg",
-                  "--apple-id",
-                  notary_user,
-                ]),
-    "--notary-arg=--password",
-    "--notary-arg",
-    notary_password,
-    "--notary-arg=--team-id",
-    "--notary-arg",
-    chrome_mac_team_id,
-    development_arg,
-    "--dmg",
-    rebase_path(signed_dmg_path),
-    "--pkg",
-    rebase_path(signed_pkg_path),
-    "--signed",
-    rebase_path("$root_out_dir/$mac_signed_output_prefix"),
-    "--outdir",
-    rebase_path("$root_out_dir/$mac_notarized_output_prefix"),
-    "--pkgdir",
-    rebase_path(packaging_dir),
-    "--mac_provisioning_profile",
-    rebase_path(provisioning_profile),
+group("create_dist_mac") {
```

Comment:
do we need `_mac` suffix here? we're already in `//brave/mac`.

Reply:
I'm getting a surprising error without the `_mac` suffix:

```
> git diff
--- a/BUILD.gn
+++ b/BUILD.gn
   if (is_mac) {
-    deps += [ "//brave/build/mac:create_dist_mac" ]
+    deps += [ "//brave/build/mac:create_dist" ]
   }
--- a/build/mac/BUILD.gn
+++ b/build/mac/BUILD.gn
-group("create_dist_mac") {
+group("create_dist") {
   deps = [
     ":copy_dmg",
     ":copy_zip",
> npm run create_dist [...]
[...]
ninja: error: unknown target 'create_dist'
```

Reply:
I see. Let's keep it as is then.